packages/core/src/components/alert/alert.tsx

```diff
@@ -45,54 +66,93 @@ export interface IAlertProps extends IProps {
     style?: React.CSSProperties;
 
     /**
-     * Handler invoked when the cancel button is clicked.
+     * Indicates how long (in milliseconds) the overlay's enter/leave transition takes.
+     * This is used by React `CSSTransition` to know when a transition completes and must match
+     * the duration of the animation in CSS. Only set this prop if you override Blueprint's default
+     * transitions with new transitions of a different length.
+     * @default 300
+     */
+    transitionDuration?: number;
+
+    /**
+     * Handler invoked when the alert is canceled. Alerts can be **canceled** in the following ways:
+     * - clicking the cancel button (if `cancelButtonText` is defined)
+     * - pressing the escape key (if `canEscapeKeyCancel` is enabled)
+     * - clicking on the overlay backdrop (if `canOutsideClickCancel` is enabled)
+     *
+     * If any of the `cancel` props are defined, then either `onCancel` or `onClose` must be defined.
      */
-    onCancel?(e: React.MouseEvent<HTMLButtonElement>): void;
+    onCancel?(evt?: React.SyntheticEvent<HTMLElement>): void;
 
     /**
-     * Handler invoked when the confirm button is clicked.
+     * Handler invoked when the confirm button is clicked. Alerts can be **confirmed** in the following ways:
+     * - clicking the confirm button
+     * - focusing on the confirm button and pressing `enter` or `space`
      */
-    onConfirm(e: React.MouseEvent<HTMLButtonElement>): void;
+    onConfirm?(evt?: React.SyntheticEvent<HTMLElement>): void;
+
+    /**
+     * Handler invoked when the Alert is confirmed or canceled; see `onConfirm` and `onCancel` for more details.
+     * First argument is `true` if confirmed, `false` otherwise.
+     * This is an alternative to defining separate `onConfirm` and `onCancel` handlers.
```

Comment:
why support 3 handlers when one will do? either remove this (leave the callbacks as-is) or remove the existing ones and only support `onClose`

Reply:
`TagInput` also has three callbacks

Reply:
alright fine 🤷‍♂️